### PR TITLE
client id middleware

### PR DESCRIPTION
**Pull request by alessandroMiceli**

### CODE REVIEW

File: http/middleware/response_header.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/pace/bricks/http/oauth2"
+	"github.com/pace/bricks/maintenance/log"
+)
+
+// ClientIDHeaderName name of the HTTP header that is used for reporting
+const ClientIDHeaderName = "Client-ID"
```
Comment (daemonfire300) [inline http/middleware/response_header.go:17]: Please check if this header is used otherwhere, i.e., whether this is a standard header that we might conflict with.

File: http/middleware/response_header.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+func (rcc *ResponseClientIDContext) AddResponseClientID(clientID string) {
+	rcc.mu.Lock()
+	rcc.clientIDs = append(rcc.clientIDs, responseClientID{
+		ClientID: clientID,
+	})
+	rcc.mu.Unlock()
+}
+
+// String formats all client IDs
+func (rcc *ResponseClientIDContext) String() string {
```
Comment (daemonfire300) [inline http/middleware/response_header.go:91]: Is this in the spirit of the issue, i.e., is this what was requested (have all client ids?)

**Comment by daemonfire300** [comment]
@alessandroMiceli ping

File: http/middleware/response_header.go (hunk 2 above)
Comment (alessandroMiceli) [reply http/middleware/response_header.go:91]: https://lab.jamit.de/pace/web/meta/-/issues/124 this is the link to the issue

**Pull request closed without merging**